src/globus_cli/login_manager.py

```diff
@@ -0,0 +1,87 @@
+import functools
+
+import click
+
+from .tokenstore import token_storage_adapter
+
+
+class LoginManager:
+    def __init__(self):
+        self._token_storage = token_storage_adapter()
+
+    def has_login(self, resource_server: str):
+        """
+        Determines if the user has a refresh token for the given
+        resource server
+        """
+        tokens = self._token_storage.get_token_data(resource_server)
+        return tokens is not None and "refresh_token" in tokens
+
+
+def requires_login(*args: str, pass_manager: bool = False):
```

Comment:
```suggestion
def requires_login(*resource_servers: str, pass_manager: bool = False):
```

Can we make this change (and then remove the assignment below)?
I think it will make this function much more readable. When I first saw it, I thought you were passing _these_ args to the decorated function, because you're overriding the same name (`args`).